docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,237 @@
+package metadata
+
+components: sources: exec: {
+	title: "Exec"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["sidecar"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.exec
+			}
+
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		mode: {
+			description: "The type of exec mechanism."
+			required:    true
+			type: string: {
+				enum: {
+					scheduled: "Scheduled exec mechanism."
+					streaming: "Streaming exec mechanism."
+				}
+				syntax: "literal"
+			}
+		}
+		command: {
+			required:    true
+			description: "The command to be run."
+			type: string: {
+				examples: ["echo", "./myscript.sh"]
+				syntax: "literal"
+			}
+		}
+		arguments: {
+			common:      false
+			description: "Array of any arguments to pass to the command."
+			required:    false
+			type: array: {
+				default: null
+				items: type: [string]: {
+					examples: ["Hello World!", "-la"]
+					syntax: "literal"
+				}
+			}
+		}
+		current_dir: {
+			common:      false
+			required:    false
+			description: "The directory from within which to run the command."
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		include_stderr: {
+			common:      false
+			description: "Include the output of stderr when generating events."
+			required:    false
+			type: bool: default: false
+		}
+		event_per_line: {
+			common:      false
+			description: "Determine if events should be generated per line."
+			required:    false
+			type: bool: default: true
+		}
+		maximum_buffer_size: {
+			common:      false
+			description: "The maximum buffer size allowed before a log event will be generated."
+			required:    false
+			type: uint: {
+				default: 1000000
+				unit:    "bytes"
+			}
+		}
+		exec_interval_secs: {
```

Comment:
We currently have a mix of config styles for these config options which are only relevant in specific "mode"s or "strategy"s. I think we are moving towards using nested maps for mode specific config, like the `aws_s3` source:

https://github.com/timberio/vector/blob/a7d3f047dbfc2ca2df1fa56ae187d504e8113c8a/docs/reference/components/sources/aws_s3.cue#L49-L124

What would you think of modifying this to move the mode specific config options under a key matching the mode (`scheduled` or `streaming`)? This would essentially be unflattening the config you flatten below.

Reply:
I have modified the logic and documentation so that the `scheduled` and `streaming` options appear nested under the relevant key e.g. `scheduled.exec_interval_secs`